File: jobs/2026-05-30__07-57-35/encore_cron_job_cleanup_go__nCsAvB6/artifacts/user/cleanup-app/cleanup/cleanup.go
package cleanup

import (
	"context"
	"time"

	"encore.dev/cron"
	"encore.dev/storage/sqldb"
)

//encore:service
type Service struct{}

var db = sqldb.NewDatabase("cleanup_db", sqldb.DatabaseConfig{
	Migrations: "migrations",
})

type Record struct {
	ID        string    `json:"id"`
	Data      string    `json:"data"`
	CreatedAt time.Time `json:"created_at"`
}

type CreateRecordRequest struct {
	ID        string    `json:"id"`
	Data      string    `json:"data"`
	CreatedAt time.Time `json:"created_at"`
}

//encore:api public method=POST path=/records
func CreateRecord(ctx context.Context, req *CreateRecordRequest) error {
	_, err := db.Exec(ctx, `INSERT INTO records (id, data, created_at) VALUES ($1, $2, $3)`, req.ID, req.Data, req.CreatedAt)
	return err
}

type ListRecordsResponse struct {
	Records []Record `json:"records"`
}

//encore:api public method=GET path=/records
func ListRecords(ctx context.Context) (*ListRecordsResponse, error) {
	rows, err := db.Query(ctx, `SELECT id, data, created_at FROM records ORDER BY created_at DESC`)
	if err != nil {
		return nil, err
	}
	defer rows.Close()

	records := make([]Record, 0)
	for rows.Next() {
		var record Record
		if err := rows.Scan(&record.ID, &record.Data, &record.CreatedAt); err != nil {
			return nil, err
		}
		records = append(records, record)
	}

	if err := rows.Err(); err != nil {
		return nil, err
	}

	return &ListRecordsResponse{Records: records}, nil
}

type CleanupResponse struct {
	Deleted int64 `json:"deleted"`
}

//encore:api public method=POST path=/cleanup
func Cleanup(ctx context.Context) (*CleanupResponse, error) {
	cutoff := time.Now().Add(-24 * time.Hour)
	result, err := db.Exec(ctx, `DELETE FROM records WHERE created_at < $1`, cutoff)
	if err != nil {
		return nil, err
	}
	deleted, err := result.RowsAffected()
	if err != nil {
		return nil, err
	}

	return &CleanupResponse{Deleted: deleted}, nil
}

var _ = cron.NewJob("cleanup-old-records", cron.JobConfig{
	Title:    "Cleanup old records",
	Endpoint: Cleanup,
	Schedule: "every 1h",
})

File: jobs/2026-07-03__12-25-47/encore_cron_job_cleanup_ts__jzmm7Zf/artifacts/user/cron-app-zrz2ku2n82/cleanup/cleanup.ts
import { api } from "encore.dev/api";
import { recordsDB } from "./db";

interface CreateRecordRequest {
  data: string;
  created_at?: string;
}

interface CreateRecordResponse {
  id: number;
}

interface RecordItem {
  id: number;
  data: string;
  created_at: string;
}

interface ListRecordsResponse {
  records: RecordItem[];
}

interface CleanupResponse {
  deleted: number;
}

export const createRecord = api(
  { method: "POST", path: "/records" },
  async (req: CreateRecordRequest): Promise<CreateRecordResponse> => {
    let createdAt: string;
    if (req.created_at) {
      createdAt = req.created_at;
    } else {
      createdAt = new Date().toISOString();
    }
    const row = await recordsDB.queryRow<{ id: number }>`
      INSERT INTO records (data, created_at)
      VALUES (${req.data}, ${createdAt}::timestamp)
      RETURNING id
    `;
    return { id: row!.id };
  }
);

export const listRecords = api(
  { method: "GET", path: "/records" },
  async (): Promise<ListRecordsResponse> => {
    const rows = await recordsDB.query<RecordItem>`
      SELECT id, data, created_at FROM records ORDER BY id
    `;
    const records: RecordItem[] = [];
    for await (const row of rows) {
      records.push({
        id: row.id,
        data: row.data,
        created_at: typeof row.created_at === 'string' ? row.created_at : new Date(row.created_at).toISOString(),
      });
    }
    return { records };
  }
);

export const cleanupRecords = api(
  { method: "POST", path: "/cleanup" },
  async (): Promise<CleanupResponse> => {
    const row = await recordsDB.queryRow<{ count: number }>`
      WITH deleted AS (
        DELETE FROM records WHERE created_at < NOW() - INTERVAL '1 hour'
        RETURNING 1
      )
      SELECT COUNT(*)::int AS count FROM deleted
    `;
    return { deleted: row?.count ?? 0 };
  }
);
